lib/dk_buttons_and_leds/dk_buttons_and_leds.c

```diff
@@ -24,33 +24,32 @@ static button_handler_t button_handler_cb;
 static atomic_t my_buttons;
 struct device *gpio_dev;
 
-static int get_buttons(u32_t *mask)
+static u32_t get_buttons(void)
 {
+	u32_t ret = 0;
 	for (size_t i = 0; i < ARRAY_SIZE(button_pins); i++) {
 		u32_t val;
 
 		if (gpio_pin_read(gpio_dev, button_pins[i], &val)) {
 			LOG_ERR("Cannot read gpio pin");
-			return -EFAULT;
+			return 0;
```

Comment:
In case of return error, we assume that the state of every button is not pressed  ?

Reply:
The error here is very unlikely. And this is quite similar it was before. We can return the current state eventually.

Reply:
I think it can be the way you did it.